db/hints/sync_point.cc

```diff
@@ -67,16 +69,30 @@ static std::vector<sync_point::shard_rps> decode_one_type_v1(uint16_t shard_coun
     return ret;
 }
 
+static uint64_t get_checksum(signed char* data, size_t size) {
```

Comment:
Nit: I think `calculate_checksum` would be a slightly better name. `get_checksum` could imply that maybe the checksum is encoded in the data and it "gets" it out of the encoded data.